src/main/kotlin/no/skatteetaten/aurora/boober/feature/S3Feature.kt

```diff
@@ -53,100 +57,96 @@ class S3Feature(
     val herkimerService: HerkimerService,
     @Value("\${application.deployment.id}") val booberApplicationdeploymentId: String,
     @Value("\${openshift.cluster}") val cluster: String,
-    @Value("\${boober.productionlevel}") val productionLevel: String
-) : S3FeatureTemplate() {
+    @Value("\${boober.productionlevel}") productionLevel: String
+) : S3FeatureTemplate(productionLevel) {
 
     override fun enable(header: AuroraDeploymentSpec): Boolean {
         return !header.isJob
     }
 
     override fun modify(adc: AuroraDeploymentSpec, resources: Set<AuroraResource>, cmd: AuroraContextCommand) {
-
-        val s3Secret = resources.find { it.resource.metadata.name == adc.s3SecretName }
-            ?.let { it.resource as Secret } ?: return
-
-        val envVars = s3Secret.createEnvVarRefs(prefix = "S3_")
-        addEnvVarsToDcContainers(resources, envVars)
+        resources.filter { it.resource.metadata.name.endsWith("s3") }
+            .map {
+                it.resource as Secret
```

Comment:
Hva skjer her hvis den ikke er en Secret?

Reply:
Kan ta en filter for å filtrere til kun Secrets